### ISSUE TITLE
508-defect-2 [FOCUS]: Focus management should be consistent 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#26416]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Trevor and Josh

## User Story or Problem Statement

As a screen reader user or a sighted keyboard user, I expect focus to be managed consistently across all search pages.

### Video demo

https://user-images.githubusercontent.com/14154792/122945674-bacb7700-d346-11eb-8d27-b5ed21bed704.mov

### <del>`❗ Launchblocker` Global header search button is not accessible to keyboard users. </del>

<del>Screen reader users will be instructed to use ctrl-option-space to activate it, but the button will not respond unless a mouse click is used.</del>

<del>Currently not responding to `enter`</del>

Fixed!

### <del> `defect-2` When searching from global header, focus goes to "skip to content", not the actual search results</del>

<del>Focus should go to "Showing 1-10 of..." so users know that the search has been activated and that results exist. "Showing 1-10 of..." should also be a `h2`, and not a `p` with `aria-live` to be consistent with the other search experiences.</del>

<img width="1124" alt="Screen Shot 2021-07-06 at 11 51 18 AM" src="https://user-images.githubusercontent.com/14154792/124630612-9ac0ab00-de50-11eb-8307-3ac807bb27e7.png">

Fixed!

### <del>`defect-2` No feedback is provided when there are no results</del>

<del>Focus is not managed post search when no feedback emerges. Focus should go to the error message that no results were found.</del>

<img width="805" alt="Screen Shot 2021-07-06 at 11 53 02 AM" src="https://user-images.githubusercontent.com/14154792/124630823-c8a5ef80-de50-11eb-8a6d-1ab98f6af655.png">

Fixed!

### <del> `defect-3` When searching from the search results page itself, focus goes to "Our top recommendations for you" </del>

<del>"Showing 1-10..." is announced using `aria-live` which is inconsistent with other search experiences that announce the number of results by sending focus to "Showing 1-10..."</del>

<img width="995" alt="Screen Shot 2021-07-06 at 11 54 28 AM" src="https://user-images.githubusercontent.com/14154792/124631030-fab75180-de50-11eb-9291-36843ffc23b1.png">

### On search, focus goes to `h2` "Showing 1-10..."  

https://user-images.githubusercontent.com/14154792/122953324-a0949780-d34c-11eb-8b5b-655947556ecd.mov

### When there are no results, focus goes to error message

https://user-images.githubusercontent.com/14154792/122954177-392b1780-d34d-11eb-929b-069e2ea4e991.mov

### On search, focus goes to group "Showing 1-10..."

https://user-images.githubusercontent.com/14154792/122954285-4c3de780-d34d-11eb-8d3c-492ab333c8b9.mov

### When there are no results, focus goes to error message

<img width="854" alt="Screen Shot 2021-06-22 at 11 32 50 AM" src="https://user-images.githubusercontent.com/14154792/122954600-945d0a00-d34d-11eb-8e4d-20cf418d2a7f.png">

Fixed!

## Acceptance Criteria

- [x] Focus management is consistent across all forms

## Proposed Solution

This will require more research and collaboration across teams.

**Comment by srsuddath** [comment]
@joshkimux We have updated the review instance to have fixes for all the launch blockers you called out, let me know when you have a chance to test em!

**Comment by joshkimux** [comment]
Updated the main ticket with fixed defects 🥳

**Comment by joshkimux** [comment]
Confirming all VA.gov search results tickets have been fixed. Find a VA Form and Resources/Results remains. I'll reassign this accordingly.

**Comment by joshkimux** [comment]
Closing this ticket out as all defects have been resolved. Headings still need adjustment to be consistent, but are documented in a separate ticket.
